src/components/families/family-form/FamilyForm.tsx

```diff
@@ -28,47 +81,95 @@ const FamilyForm = ({
   guestDynamicFields,
   parentDynamicFields,
   onChange,
-}: Props) => (
-  <>
-    <Typography component="h3" variant="h5">
-      Basic information
-    </Typography>
-    <FamilyParentFields
-      dynamicFields={parentDynamicFields}
-      family={family}
-      onChange={(value) => onChange({ ...family, ...value })}
-    />
-
-    <Typography component="h3" variant="h5">
-      Children
-    </Typography>
-    <StudentFields
-      dynamicFields={childDynamicFields}
-      onChange={(value) =>
-        onChange({
-          ...family,
-          children: [{ ...family.children[0], ...value }],
-        })
-      }
-      role={StudentRole.CHILD}
-      student={family.children[0]}
-    />
-
-    <Typography component="h3" variant="h5">
-      Family members
-    </Typography>
-    <StudentFields
-      dynamicFields={guestDynamicFields}
-      onChange={(value) =>
-        onChange({
-          ...family,
-          guests: [{ ...family.guests[0], ...value }],
-        })
-      }
-      role={StudentRole.GUEST}
-      student={family.guests[0]}
-    />
-  </>
-);
+}: Props) => {
+  const addStudent = (role: StudentRole.GUEST | StudentRole.CHILD): void => {
+    if (role === StudentRole.CHILD) {
+      onChange({
+        ...family,
+        children: [
+          ...family.children,
+          { ...defaultStudentData, index: generateKey(role) },
+        ],
+      });
+    } else {
+      onChange({
+        ...family,
+        guests: [
+          ...family.guests,
+          { ...defaultStudentData, index: generateKey(role) },
+        ],
+      });
+    }
+  };
+
+  const updateStudent = (
+    index: number,
+    data: StudentRequest,
+    role: StudentRole.GUEST | StudentRole.CHILD
+  ): void => {
+    const students =
+      role === StudentRole.CHILD ? [...family.children] : [...family.guests];
+    students[index] = { ...students[index], ...data };
+    if (role === StudentRole.CHILD) {
+      onChange({ ...family, children: students });
+    } else {
+      onChange({ ...family, guests: students });
+    }
+  };
+
+  const deleteStudent = (
+    index: number,
+    role: StudentRole.CHILD | StudentRole.GUEST
+  ): void => {
+    const data = role === StudentRole.CHILD ? family.children : family.guests;
+    const students = data.filter((e) => e.index !== index);
+    if (role === StudentRole.CHILD) {
+      onChange({ ...family, children: [...students] });
+    } else {
+      onChange({ ...family, guests: [...students] });
+    }
+  };
```

Comment:
are we able to use object spread @L124? This would make the logic between updateStudent() and deleteStudent() a bit more consistent.